concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -586,67 +760,91 @@ writeChunks
+                    void $ writeBlockOut serializedBlock
+                    -- In case there wasn't a @FinalizationRecord@ present in the block
+                    -- we carry over the latest seen @FinalizationIndex@.
+                    continue count height $ fromMaybe lastFinRecIdx finalizationRecordIndexM
+        GetBlockAtV1 f ->
+            f height >>= \case
+                Nothing -> return (count, 0)
+                Just serializedBlock -> do
+                    void $ writeBlockOut serializedBlock
+                    -- We simply set the last @FinalizationRecord@ index to 0,
+                    -- as they are not a concept for 'ConsensusV1'.
+                    continue count height 0
+    writeBlockOut :: MonadIO m => BS.ByteString -> m ()
+    writeBlockOut serializedBlock = do
+        let len = fromIntegral $ BS.length serializedBlock
+        liftIO $ do
+            BS.hPut hdl $ runPut $ putWord64be len
+            BS.hPut hdl serializedBlock
+    continue count height finalizationRecordIndex =
+        if count < chunkSize
+            then ebtc (height + 1) (count + 1) finalizationRecordIndex
+            else return (count, finalizationRecordIndex)
 
 -- |Export all finalization records with indices above `dbsLastFinIndex` to a chunk
+-- Note. For 'ConsensusV1' this function will not write anything to the file.
 exportFinRecsToChunk ::
-    forall pv m.
-    (MonadIO m, MonadState (DBState pv) m, MonadCatch m) =>
+    (MonadIO m, IsConsensusParametersVersion cpv) =>
     -- |Handle to export to
     Handle ->
+    -- |Last finalization record index
+    FinalizationIndex ->
+    -- |Action for getting the finalization record at a given index.
+    GetFinalizationRecordAt cpv m ->
     -- |Number of exported finalization records
     m Word64
-exportFinRecsToChunk hdl = exportFinRecsFrom 0 . (+ 1) =<< use dbsLastFinIndex
+exportFinRecsToChunk hdl finRecIdx getFinalizationRecordAt = exportFinRecsFrom (0 :: Word64) (1 + finRecIdx)
   where
     exportFinRecsFrom count finRecIndex =
-        resizeOnResized (readFinalizationRecord finRecIndex) >>= \case
-            -- if there's no finalization record with an index after the last exported one,
-            -- there are no more finalization records to export
-            Nothing -> return count
-            Just fr -> do
-                let serializedFr = runPut $ putVersionedFinalizationRecordV0 fr
-                    len = fromIntegral $ BS.length serializedFr
-                liftIO $ do
-                    BS.hPut hdl $ runPut $ putWord64be len
-                    BS.hPut hdl serializedFr
-                exportFinRecsFrom (count + 1) (finRecIndex + 1)
+        case getFinalizationRecordAt of
+            -- Write out the @FinalizationRecord@s above the last @FinalizationIndex@.
+            -- We terminate the loop when there is the 'GetFinalizationRecordAtV0' returns
+            -- 'Nothing'.
+            GetFinalizationRecordAtV0 f -> do
```

Comment:
Something is a bit off in this sentence.

Reply:
fixed.